vllm/executor/ray_gpu_executor.py

```diff
@@ -234,6 +243,19 @@ def _init_workers_ray(self, placement_group: "PlacementGroup",
                           max_concurrent_workers=self.parallel_config.
                           max_parallel_loading_workers)
 
+        if self.use_ray_spmd_worker:
+            for pp_rank in range(self.parallel_config.pipeline_parallel_size):
+                self.pp_tp_workers.append([])
+                for tp_rank in range(
+                        self.parallel_config.tensor_parallel_size):
+                    # PP=2, TP=4
+                    # pp_tp_workers = [[0, 1, 2, 3], [4, 5, 6, 7]]
+                    rank = (pp_rank * self.parallel_config.tensor_parallel_size
```

Comment:
QQ: is there any way to reuse logic inside https://github.com/vllm-project/vllm/blob/7f8d612d24c66e9b5f8c0aa6cb562e853e9523a0/vllm/distributed/parallel_state.py#L754?